pkg/controller/storagecluster/reconcile.go

```diff
@@ -814,6 +888,26 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string, nodeCount int, r
 	return cephCluster
 }
 
+func newExternalCephCluster(sc *ocsv1.StorageCluster, cephImage string) *cephv1.CephCluster {
+	labels := map[string]string{
+		"app": sc.Name,
+	}
+	externalCephCluster := &cephv1.CephCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForExternalCephCluster(sc),
```

Comment:
Either use `generateNameForExternalCephCluster` or `generateNameForCephCluster`.
Currently, you've mixed both the implementations.

Reply:
Removed `generateNameForExternalCephCluster`, now using only `generateNameForCephCluster` method...